tests/ci/attach_gdb.lib

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+function attach_gdb_to_clickhouse()
+{
+    # Set follow-fork-mode to parent, because we attach to clickhouse-server, not to watchdog
+    # and clickhouse-server can do fork-exec, for example, to run some bridge.
+    # Do not set nostop noprint for all signals, because some it may cause gdb to hang,
+    # explicitly ignore non-fatal signals that are used by server.
+    # Number of SIGRTMIN can be determined only in runtime.
+    RTMIN=$(kill -l SIGRTMIN)
+  echo "
+set follow-fork-mode parent
+handle SIGHUP nostop noprint pass
+handle SIGINT nostop noprint pass
+handle SIGQUIT nostop noprint pass
+handle SIGPIPE nostop noprint pass
+handle SIGTERM nostop noprint pass
+handle SIGUSR1 nostop noprint pass
+handle SIGUSR2 nostop noprint pass
+handle SIG$RTMIN nostop noprint pass
+info signals
+continue
+backtrace full
+thread apply all backtrace full
+info registers
+disassemble /s
+up
+disassemble /s
+up
+disassemble /s
+p \"done\"
+detach
+quit
+" > script.gdb
+
+    # FIXME Hung check may work incorrectly because of attached gdb
+    # We cannot attach another gdb to get stacktraces if some queries hung
+    gdb -batch -command script.gdb -p "$(cat /var/run/clickhouse-server/clickhouse-server.pid)" | ts '%Y-%m-%d %H:%M:%S' >> /test_output/gdb.log &
```

Comment:
We need to kill this gdb before calling `get_stacktraces_from_gdb` in `clickhouse-test` if some queries hung (of course it would be much better to avoid starting another instance of `gdb` and just send some signal from `clickhouse-test` to make this instance stop waiting on `continue` and print the stacktraces, but I failed to do it). See also:
https://github.com/ClickHouse/ClickHouse/blob/48b23dd012d95d2b2d477d4fa448f4e88f628c8e/tests/ci/stress_tests.lib#L154
https://github.com/ClickHouse/ClickHouse/blob/0598bfd79e6518c9432e341af100a31649f457e0/tests/ci/stress.py#L122-L126

Reply:
Will it be ok to just kill gdb in `clickhouse-test` inside `get_stacktraces_from_gdb` function before starting another gdb?

Reply:
Yep